### PR TITLE
Extract common to nuget

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This is the first step during the extraction of Common projects into a separate solution with deployment to multiple packages. 

Next step is to remove Common code from the Fitnet solution and references these packages. This will be handled in a separate branch.

### CODE REVIEW

File: Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IApplicationMemory.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace EvolutionaryArchitecture.Fitnet.Common.IntegrationTests.TestEngine;
+
+public interface IApplicationFactory
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IApplicationMemory.cs:3]: File has different name comparing to interface name
image

Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IApplicationMemory.cs:3]: Probably we don't use this interface anymore after get rid of in memory event bus.

image

File: Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IntegrationEvents/Handlers/IntegrationEventHandlerScope.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+namespace EvolutionaryArchitecture.Fitnet.Common.IntegrationTests.TestEngine.IntegrationEvents.Handlers;
+
+using Infrastructure.Events;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+
+public sealed class IntegrationEventHandlerScope<TIntegrationEvent> : IDisposable
```
Comment (kamilbaczek) [inline Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IntegrationEvents/Handlers/IntegrationEventHandlerScope.cs:7]: Probably we can remove it

File: Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IApplicationMemory.cs (hunk 1 above)
Comment (meaboutsoftware) [reply Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IApplicationMemory.cs:3]: I will remove it (as you are right, we will not use it anymore)

Comment (meaboutsoftware) [reply Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IApplicationMemory.cs:3]: Removed

File: Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IntegrationEvents/Handlers/IntegrationEventHandlerScope.cs (hunk 2 above)
Comment (meaboutsoftware) [reply Chapter-3-microservice-extraction/Fitnet.Common/Fitnet.Common.IntegrationTests/TestEngine/IntegrationEvents/Handlers/IntegrationEventHandlerScope.cs:7]: Removed

**Review by kamilbaczek — APPROVED** [review]
Great job! Thanks

**Pull request merged**